### ISSUE TITLE
Emotion Classifiers

### ISSUE THREAD

**Issue by manzar96** [facebookresearch/EmpatheticDialogues#32]
Hi,
I would like to ask in which dataset the classifier was pretrained and what was the architecture of the classifier.

**Comment by EricMichaelSmith** [comment]
Hi! A fastText model (https://fasttext.cc/) was used to prepend emotion and topic labels.

**Comment by manzar96** [comment]
Excuse me but I can't understand the concept exactly. The pretrained classifier was used to extract top k sentiments and the you prepended them to the input. (Am I right?) So that classifier was trained given the emotion labels of Empathetic Dataset or in another dataset?

**Comment by EricMichaelSmith** [comment]
I didn't train this particular classifier, but from the paper it looks like it was trained on the ED emotion labels: "To provide emotion signal, we train a classifier to predict the emotion label from the description of the situation written by the Speaker before the dialogue for the training set dialogues of ED (EMOPREPEND-1)."

**Comment by manzar96** [comment]
ok! thank you a lot!

**Comment by EricMichaelSmith** [comment]
Sure thing :)

**Comment by manzar96** [comment]
Can you please provide us the pre-trained classifier?

**Comment by EricMichaelSmith** [comment]
Hi! See notes for this in the README: do `wget https://dl.fbaipublicfiles.com/parlai/empatheticdialogues/models/fasttext_empathetic_dialogues.mdl`.

**Comment by manzar96** [comment]
Thanks for your help! I would like to wrap this model and use it on my own code in another project (using pytorch). However I can not understand if that's possible as` fasttext.load_model()` returns a FastText object.
Can you please provide me some details to handle this issue?

**Comment by EricMichaelSmith** [comment]
Sorry for the delay! So, I've never tried porting FastText to PyTorch but I'm sure it's possible - Googling "fasttext to pytorch" gives lots of results that might help, for instance. Good luck!
